pandas/core/util/hashing.py

```diff
@@ -324,8 +324,8 @@ def _hash_ndarray(
             )
 
             codes, categories = factorize(vals, sort=False)
-            dtype = CategoricalDtype(categories=Index(categories), ordered=False)
-            cat = Categorical._simple_new(codes, dtype)
+            dtype = CategoricalDtype(categories=Index(categories), ordered=False)  # type: ignore[assignment]
+            cat = Categorical._simple_new(codes, dtype)  # type: ignore[arg-type]
```

Comment:
```suggestion
            tdtype = CategoricalDtype(categories=Index(categories), ordered=False)  
            cat = Categorical._simple_new(codes, tdtype)  
```
That will get rid of the ignores and still work